Add reading time for entries (fixes #548)

Mexx62 opened the pull request:
Hey!  

I wanted a reading time estimate for articles, as asked in the issue #548. Here's the PR implementing it.  
I'm new to Go, so feel free to make any remark about it.
To estimate the reading time, I take the number of words, divide it by 265 WPM (average), and round it up.

As I can't speak any language fluently except French and English, I didn't translate the settings label. Do you want me to Google Translate it and wait for someone to PR it? Or do we merge it with the english label?

Regards,
Maxime

Mexx62 commented:
> Yes, you should include the translations for each language. You can use Google Translate or similar.

I added the translations for each language via Google Translate.

> Please, do not change the indentation of language files in JSON otherwise it's impossible to review.

Oops, sorry for that, my IDE wasn't well configured. It's fixed now.

> You should also include generated files in your PR. As you can see, the integration tests are failing.

Added.  

Thanks for this first round of remarks!

fguillot commented on template/html/common/item_meta.html:
```diff
@@ -7,6 +7,11 @@
         </li>
+        {{ if .user.ShowReadingTime }}
+        <li>
+            <span>{{ timeToRead .entry.Content }} min read</span>
```
I just noticed that `min read` is not translated.

Mexx62 replied:
Do you think it would it be better with `%d min read` or `estimated reading time : %d minutes`?

fguillot replied:
The shorter version seems better: `%d min read`

Mexx62 commented:
Oh I didn't see that, thanks for the review! I'm fixing it this evening

15 juil. 2020 07:04:33 Frédéric Guillot <notifications@github.com>:

> @fguillot commented on this pull request.
> 
> 
> ----------------------------------------
> In template/html/common/item_meta.html[https://github.com/miniflux/miniflux/pull/726#discussion_r454793560]:
> 
>> @@ -7,6 +7,11 @@ <li> <time datetime="{{ isodate .entry.Date }}" title="{{ isodate .entry.Date }}">{{ elapsed .user.Timezone .entry.Date }}</time> </li> +        {{ if .user.ShowReadingTime }} +        <li> +            <span>{{ timeToRead .entry.Content }} min read</span>I just noticed that min read is not translated.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub[https://github.com/miniflux/miniflux/pull/726#pullrequestreview-448630813], or unsubscribe[https://github.com/notifications/unsubscribe-auth/AB5RLNN3GSESEBYRMNK4FDDR3U2FZANCNFSM4OV7WXHA]. [https://github.com/notifications/beacon/AB5RLNMVAXYEXTFVJFFBV4LR3U2FZA5CNFSM4OV7WXHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGODK6ZAHI.gif]
>

fguillot commented:
Another contributor added the Brazilian translation, you should update this PR based on the latest changes to avoid conflicts.

Mexx62 commented:
> Another contributor added the Brazilian translation, you should update this PR based on the latest changes to avoid conflicts.

I picked his translation and added my new sentences in it. The diff won't be very legible with this file though because I changed the formatting to conform to the other translations (4 spaces instead of 2).  

@pedrolucasp Could you have a look to the new sentences I added please? (this comes from Google Translate to Portuguese)
```json
"entry.estimated_reading_time": [
        "%d minuto lido",
        "%d minutos lidos"
],
"form.prefs.label.show_reading_time": "Mostrar tempo estimado de leitura de artigos",
```

fguillot commented:
Merged too fast. I didn't see the comment. I will let people update the translations in another PR if not accurate.

pedrolucasp commented:
Sorry, I've just seen this notification now. I'll send a patch later updating the translation properly. 

Thanks for the heads up!
